src/sql/workbench/services/objectExplorer/browser/dragAndDropController.ts

```diff
@@ -29,9 +31,11 @@ export class ServerTreeDragAndDrop implements IDragAndDrop {
 	public getDragURI(tree: ITree, element: any): string {
 		if (element instanceof ConnectionProfile) {
 			return (<ConnectionProfile>element).id;
-		}
-		else if (element instanceof ConnectionProfileGroup) {
+		} else if (element instanceof ConnectionProfileGroup) {
 			return (<ConnectionProfileGroup>element).id;
+
```

Comment:
What I would suggest here is to have all the changes in the `sql` side. You can get rid of the changes on the `vs` side by actually sending in the data by changing the `DragAndDrop` class right here. Since you are already sending the URI, you can perhaps have a dictionary that maps URI's to elements here. So, without changing the inner implementation, you can get the data directly by changing the `drop` function here to retrieve the data from the dictionary based on the URI. 
A good way to realize that this wasn't the correct way to do this is by observing that you had to change the interface in order for it to work. Usually, you'd never have to change an interface, especially on the `vs` side 😄